pkg/apis/harvesterhci.io/v1beta1/upgrade.go

```diff
@@ -43,6 +46,11 @@ type UpgradeSpec struct {
 	// +optional
 	// +kubebuilder:default:=true
 	LogEnabled bool `json:"logEnabled" default:"true"`
+
+	// Specifies the image to be used for upgrade operations.
+	// Please note that this field is intended for development and testing purposes only.
+	// +optional
+	UpgradeImage *string `json:"upgradeImage,omitempty"`
```

Comment:
it is mainly for developing purpose, would prefer to use an label/annotation to to overwrite

normally when building on `harvester/harvester`, it generates:

`harvester,harvester-webhook, update the harvester,harvester-crd managedchart version, upgrade-repo, upgrade`,  sololy to replace the `upgrade` image is in high risk of unconsistency, only fits for dev mode